### PR TITLE
ci: fix flaky tests for Apple Game Center authentication

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

CI steps often fail due to the url https://static.gc.apple.com/public-key/gc-prod-4.cer being often unavailable.

Related issue: #7947

### Approach

Store a hard-coded cache of the valid certificate to emulate Apple's valid server response

### TODOs before merging

- [x] Add tests
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Review by Moumouls — DISMISSED** [review]
LGTM, i was not able to download the cert on my side

File: spec/AuthenticationAdapters.spec.js [hunk 1]
```diff
@@ -1652,7 +1652,7 @@ describe('apple signin auth adapter', () => {
...
...cBcFkJVF6+Tqvkg7MKJTM\nIOYPPQtPF3AZYPsbUoRKAD7/JXrxxOSVJ7vU1mP77tYG8TcUteZ3sAwvt2dkRbm7\nZO6DcmSggv1Dg4k3goNw4GYyCY4Z2/8JSmsQ80Iv/UOOwynpBziEeZmJ4uck6zlA\n17cDkH48LBpKylaqthym5bFs9gj11pto7mvyb5BTcVuohwi6qosvbs/4VGbC2Nsz\nie416nUZfv+xxoXH995gxR2mw5cDdeCew7pSKxEhvYjT2nVdQF0q/hnPMFnOaEyT\nq79n3gwFXyt0dy8eP6KBF7EW9J6b7ubu/j7h+tQfxPM+gTXOBQIDAQABo4IBPjCC\nATowCQYDVR0TBAIwADAOBgNVHQ8BAf8EBAMCB4AwEwYDVR0lBAwwCgYIKwYBBQUH\nAwMwYQYDVR0gBFowWDBWBgZngQwBBAEwTDAjBggrBgEFBQcCARYXaHR0cHM6Ly9k\nLnN5bWNiLmNvbS9jcHMwJQYIKwYBBQUHAgIwGQwXaHR0cHM6Ly9kLnN5bWNiLmNv\nbS9ycGEwHwYDVR0jBBgwFoAUljtT8Hkzl699g+8uK8zKt4YecmYwKwYDVR0fBCQw\nIjAgoB6gHIYaaHR0cDovL3N2LnN5bWNiLmNvbS9zdi5jcmwwVwYIKwYBBQUHAQEE\nSzBJMB8GCCsGAQUFBzABhhNodHRwOi8vc3Yuc3ltY2QuY29tMCYGCCsGAQUFBzAC\nhhpodHRwOi8vc3Yuc3ltY2IuY29tL3N2LmNydDANBgkqhkiG9w0BAQsFAAOCAQEA\nI/j/PcCNPebSAGrcqSFBSa2mmbusOX01eVBg8X0G/z8Z+ZWUfGFzDG0GQf89MPxV\nwoec+nZuqui7o9Bg8s8JbHV0TC52X14CbTj9w/qBF748WbH9gAaTkrJYPm+MlNhu\ntjEuQdNl/YXVMvQW4O8UMHTi09GyJQ0NC4q92Wxvx1m/qzjvTLvrXHGQ9pEHhPyz\nvfBLxQkWpNoCNKU7UeESyH06XOrGc9MsII9deeKsDJp9a0jtx+pP4MFVtFME9SSQ\ntMBs0It7WwEf7qcRLpialxKwY2EzQ9g4WnANHqo18PrDBE10TFpZPzUh7JhMViVr\nEEbl0YdElmF8Hlamah/yNw==\n-----END CERTIFICATE-----\n`;
```
Comment (mtrezza) [inline spec/AuthenticationAdapters.spec.js]: Should we move this certificate to an external file in case we also need a cert for other tests?

**Comment by mtrezza** [comment]
Renamed the PR title from `fix: ...` to `ci: ...` since this is not a product bug fix but a CI fix; it won't appear in the changelog as it's not relevant for the product itself.

**Review by mtrezza — COMMENTED** [review]
Is this certificate in any way specific to game center because it's named that way? If not and we want to make it available as a general certificate for testing maybe we should give it a more neutral name, or add it's properties to the name to make it more descriptive.

**Comment by dblythy** [comment]
It’s specific to Game Center. It’s the certificate that is available at https://static.gc.apple.com/public-key/gc-prod-4.cer but the apple servers are now flaky so storing it ourselves is the best way to circumvent

**Comment by mtrezza** [comment]
Got it, so this is ready then.

**Review by mtrezza — APPROVED** [review]

**Pull request merged**